pinot-segment-local/src/main/java/org/apache/pinot/segment/local/utils/LuceneTextIndexUtils.java

```diff
@@ -72,4 +81,202 @@ public static Query convertToMultiTermSpanQuery(Query query) {
     LOGGER.debug("The phrase query {} is re-written as {}", query, spanNearQuery);
     return spanNearQuery;
   }
+
+  /**
+   * Parses search options from a search query string.
+   * The options can be specified anywhere in the query using __OPTIONS(...) syntax.
+   * For example: "term1 __OPTIONS(parser=CLASSIC) AND term2 __OPTIONS(analyzer=STANDARD)"
+   * Options from later occurrences will override earlier ones.
+   *
+   * @param searchQuery The search query string
+   * @return A Map.Entry containing the cleaned search term and options map, or null if no options found
+   */
+  public static Map.Entry<String, Map<String, String>> parseOptionsFromSearchString(String searchQuery) {
+    try {
+      // Early check for __OPTIONS to avoid unnecessary processing
+      if (searchQuery == null || !searchQuery.contains("__OPTIONS")) {
+        return null;
+      }
+
+      // Pattern to match __OPTIONS(...) with word boundaries
+      // (?<!\\\\) - negative lookbehind to avoid escaped __OPTIONS
+      // \\b - word boundary to ensure __OPTIONS is standalone
+      // __OPTIONS\\((.*?)\\) - non-greedy match for content inside parentheses
+      Pattern pattern = Pattern.compile("(?<!\\\\)\\b__OPTIONS\\((.*?)\\)");
```

Comment:
Is this regexp match expensive? We have run into similar performance issue before on query option. I'd also suggest keeping the options as a separate argument. You can find example of passing parameters as separate argument in some complex aggregate functions, such as `DistinctCountOffHeapAggregationFunction`